.github/workflows/prerelease.yml

```diff
@@ -0,0 +1,151 @@
+name: Prerelease pipeline
+
+on:
+  release:
+    types:
+      - prereleased
+    tags:
+      - 'v*'
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GO_VERSION: '1.9.7'
+  GPG_MAIL: 'infrastructure-eng@newrelic.com'
+  GPG_PASSPHRASE: ${{ secrets.OHAI_GPG_PASSPHRASE }}
+  GPG_PRIVATE_KEY_BASE64: ${{ secrets.OHAI_GPG_PRIVATE_KEY_BASE64 }} # base64 encoded
+  INTEGRATION: "snmp"
+  ORIGINAL_REPO_NAME: 'newrelic/nri-snmp'
+  PFX_CERTIFICATE_BASE64: ${{ secrets.OHAI_PFX_CERTIFICATE_BASE64 }} # base64 encoded
+  PFX_CERTIFICATE_DESCRIPTION: 'New Relic'
+  PFX_PASSPHRASE:  ${{ secrets.OHAI_PFX_PASSPHRASE }}
+  REPO_FULL_NAME: ${{ github.event.repository.full_name }}
+  SLACK_TOKEN: ${{ secrets.OHAI_SLACK_TOKEN }}
+  SLACK_CHANNEL: ${{ secrets.OHAI_SLACK_CHANNEL }}
+  SNYK_TOKEN: ${{ secrets.COREINT_SNYK_TOKEN }}
+  TAG: ${{ github.event.release.tag_name }}
+  DOCKER_HUB_ID: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+  DOCKER_HUB_PASSWORD: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+
+jobs:
+
+  validate:
+    name: Validate code via linters
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ env.DOCKER_HUB_ID }}
+          password: ${{ env.DOCKER_HUB_PASSWORD }}
+      - name: Validate code
+        run: make ci/validate
+
+  snyk:
+    name: Run security checks via snyk
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ env.DOCKER_HUB_ID }}
+          password: ${{ env.DOCKER_HUB_PASSWORD }}
```

Comment:
If I'm not mistaken, the `docker-login` action already writes credentials to disk, so there is no need for `DOCKER_HUB_ID` and `DOCKER_HUB_PASSWORD` to be exported as env vars. While it is not a problem, I think it's good practice to avoid exposing sensitive info when possible.

Reply:
@roobre the documentation does show using the username and password like that (and also using secrets directly like I showed above) so where did you see that about writing credentials to disk?

Reply:
About env vars, we use them to avoid multiple place changes when we need to change any secret name. For example with migration to org_level secrets.

Reply:
@arvdias If I'm not mistaken that's the usual behavior of `docker login`, it just stores some kind of token in `~/.docker/config.json` (or any other keyring it is configured to use, not sure about the GHA environment).

What I meant is that I think it is better to use secrets directly (as your proposed) for the docker login action, rather than setting an env var, which is exposed to the whole pipeline. I am not entirely sure that the second is more secure, as a malicious program could likely scrape credentials from disk, but I do think it is better practice.

That being said, using an intermediate env var for aliasing has some advantages as well, as @brushknight said.

I lean slightly towards using the secrets directly but I won't push hard for it.

Reply:
I think there is no security concern, as those env vars created for each job and each job is run on new VM. So there is no difference if we pass it through the env vars or secrets imho.

Feel free to change it if you think there will be security concerns.